File: Algorithms/p406_Queue_Reconstruction_by_Height_1.cpp
class Solution {
public:
    vector<pair<int, int>> reconstructQueue(vector<pair<int, int>>& people) {
        sort(people.begin(),people.end(),[](pair<int,int>& a, pair<int,int>& b){ return a.first<b.first || a.first==b.first&&a.second>b.second; });
        int n = people.size();
        vector<pair<int, int>> res(n);
        vector<int> used(n,0);
        for(int i=0;i<n;++i){
            int cnt = people[i].second;
            int j = 0;
            while(1){
                if(used[j]==0){
                    --cnt;
                    if(cnt<0) break;
                }
                ++j;
            }
            res[j] = people[i];
            used[j] = 1;
        }
        return res;
    }
};

File: Algorithms/p406_Queue_Reconstruction_by_Height_1.py
class Solution(object):
    def reconstructQueue(self, people):
        """
        :type people: List[List[int]]
        :rtype: List[List[int]]
        """
        n = len(people)
        res = [None]*n
        used = [False]*n
        for h,k in sorted(people,key=lambda (h,k):(h,-k)):
            cnt = k
            j = 0
            while True:
                if not used[j]:
                    cnt -= 1
                    if cnt<0:
                        break
                j += 1
            res[j] = [h,k]
            used[j] = True
        return res
